server/API/REST.php

```diff
@@ -0,0 +1,170 @@
+
+define("API_PATH", $_SERVER["SCRIPT_NAME"]."/api");
+
+/* Turning warning and notices into exceptions */
+
+set_error_handler(function($errno, $errstr, $errfile, $errline) {
+    // error was suppressed with the @-operator
+    if (0 === error_reporting()) {
+        return false;
+    }
+
+    throw new ErrorException($errstr, 0, $errno, $errfile, $errline);
+});
+
+/* Utilities */
+
+if(!function_exists("check_login")){
+	/**
+	 * Check if a user is logged
+	 * 
+	 * @return bool true if is logged, false otherwise
+	 */
+	function check_login(){
+			$pdo = new PDO("sqlite:../db.sqlite");
+
+			$stmt = $pdo->prepare("SELECT ID, username, password, type FROM users WHERE ID = :userId");
+			$stmt->bindValue(":userId", $_SESSION["user_id"]);
+
+			if(!$stmt->execute()){
+				throw new Error($stmt->errorInfo(), $stmt->errorCode());
+			}
+
+			$user_data = $stmt->fetch();
+
+			return $_SESSION["nonce"] == md5(serialize($user_data));
+	}
+}
+
+
+/* Functions that implements endpoints */
+
+if(!function_exists("do_login")){
+	function do_login($vars){
+		try{
+			$pdo = new PDO("sqlite:../db.sqlite");
+
+			$stmt = $pdo->prepare("SELECT ID, username, password, type FROM users WHERE username = :username");
+			$stmt->bindValue(":username", $_POST["username"]);
+
+			if(!$stmt->execute()){
+				throw new Error($stmt->errorInfo(), $stmt->errorCode());
+			}
+
+			$user_data = $stmt->fetch();
+
+			if(!password_verify($_POST["password"], $user_data["password"])){
+				http_response_code(403);
+				echo json_encode(array('success' => false));
+				return;
+			}
+			else{
+				//TODO da trasformare in un vero nonce da salvare nel db
+				$nonce = md5(serialize($user_data));
+
+				$_SESSION['user_id'] = $user_data["ID"];
+				$_SESSION['nonce'] = $nonce;
+
+				echo json_encode(array("success" => true, "userId" => 
+				$user_data['ID'], 'type' => $user_data['type']));
+
+				return;
+			}
+		}
+		catch(Exception $e){
+			echo json_encode(array('success' => false, 'reason' => $e->getMessage()));
+		}
+	}
+}
+
+if(!function_exists('am_i_logged')){
+	function am_i_logged(){
+		if(!check_login()){
+			http_response_code(403);
+			echo json_encode(array('success' => true, 'loggedIn' => false));
+		}
+		else
+			echo json_encode(array('success' => true, 'loggedIn' => true));
+	}
+}
+
+if(!function_exists('do_logout')){
+	function do_logout($vars){
+		unset($_SESSION['nonce']);
+		unset($_SESSION['user_id']);
+
+		echo json_encode(array('success' => true));
+	}
+}
+
+/*Documentation for FastRoute can be found here: https://github.com/nikic/FastRoute */
+
+/*Constants to define option for the routes*/
+
+define("NEED_AUTH", 0); //if set, the rout needs $_SESSION['nonce'] to be set and valid
+
+//define the routes
+$dispatcher = FastRoute\simpleDispatcher(function (FastRoute\RouteCollector $r) {
+	$r->addRoute('POST', API_PATH . '/login', 'do_login');
+	$r->addRoute('POST', API_PATH . '/logout', ['do_logout', NEED_AUTH]);
+});
```

Comment:
I think you forgot the route to check the login
```suggestion
$dispatcher = FastRoute\simpleDispatcher(function (FastRoute\RouteCollector $r) {
	$r->addRoute('POST', API_PATH . '/login', 'do_login');
	$r->addRoute(''GET', API_PATH . '/logged', 'am_i_logged');
	$r->addRoute('POST', API_PATH . '/logout', ['do_logout', NEED_AUTH]);
});
```